main.tf

```
@@ -1,9 +1,13 @@
 data "aws_caller_identity" "current" {}
 
+locals {
+  kms_key_id = var.use_encryption == true ? coalesce(var.kms_master_key_id, "alias/aws/sns") : ""
```

```suggestion
  kms_key_id = var. encryption_enabled ? var.kms_master_key_id : ""
```

No need to `coalesce` since the default of `var.kms_master_key_id` is `alias/aws/sns`.
If you don't want to use encryption, just don't change the variable at all.
```suggestion
  kms_key_id = var.use_encryption == true ? coalesce(var.kms_master_key_id, "alias/aws/sns") : ""
```

---

@nitrocode , @aknysh - updated, please review!